### PR TITLE
PMM-14267 Fix persistence of changed listen ports

### PR DESCRIPTION
**Pull request by ademidoff**
PMM-14267

Link to the Feature Build: SUBMODULES-4217

This pull request introduces a mechanism for immediate VictoriaMetrics configuration updates when an agent's listen port changes, preventing stale port scraping. It adds a new method to the `prometheusService` interface, updates the handler logic to detect port changes, and provides comprehensive unit tests for this detection. Additionally, mock implementations and interface definitions across multiple packages are updated to support the new method.

Immediate VictoriaMetrics configuration update on port changes:

* Added `ForceConfigurationUpdate(ctx context.Context) error` to the `prometheusService` interface in `managed/services/agents/deps.go`, `managed/services/inventory/deps.go`, `managed/services/management/deps.go`, and `managed/services/server/deps.go`, along with documentation explaining its purpose for critical updates like port changes. [[1]](diffhunk://#diff-6ebcda5e78694f3f22d9ae4ce8d981f1ac132a6ba15822962d813eb495276878R35-R37) [[2]](diffhunk://#diff-9257e49b60cb445a48d10eaca1dcde7206412eb113f36e8f3f066f2b2eb0c0e3R51-R53) [[3]](diffhunk://#diff-0a3d788d443e4415598f2d07830f71cc604452c9cb2f9222f837a48906bf7a1dR50-R52) [[4]](diffhunk://#diff-2f2772dcd8509b7d183062bdf5c31b3b723bdee73c7d0d0fa09be63d7a2d3420R45-R47)
* Updated agent state change handling in `managed/services/agents/handler.go` to detect listen port changes using the new `checkPortChanged` function, and trigger `ForceConfigurationUpdate` for VictoriaMetrics when a port change is detected. [[1]](diffhunk://#diff-dc09f2977aa232c4d51ad9915f2d40b9377bdc34017551f29e680ab282a563e1R177-R180) [[2]](diffhunk://#diff-dc09f2977aa232c4d51ad9915f2d40b9377bdc34017551f29e680ab282a563e1R190-R196) [[3]](diffhunk://#diff-dc09f2977aa232c4d51ad9915f2d40b9377bdc34017551f29e680ab282a563e1R215-R226) [[4]](diffhunk://#diff-dc09f2977aa232c4d51ad9915f2d40b9377bdc34017551f29e680ab282a563e1R239-R251)
* Replaced direct agent reloads with `models.FindAgentByID` for consistency and reliability in agent lookup. [[1]](diffhunk://#diff-b7274f0591756f87d4c6d0dde777d34a09753e4632c48c0c1f7e1bfb7edfeb01R39) [[2]](diffhunk://#diff-dc09f2977aa232c4d51ad9915f2d40b9377bdc34017551f29e680ab282a563e1L230-R264)

Testing and mocks:

* Added a comprehensive unit test for `checkPortChanged` in `managed/services/agents/handler_test.go`, covering edge cases such as agent not found, old port zero, port equality, port difference, and type conversion.
* Updated mock implementations in `managed/services/inventory/mock_prometheus_service_test.go`, `managed/services/management/mock_prometheus_service_test.go`, and `managed/services/server/mock_prometheus_service_test.go` to support the new `ForceConfigurationUpdate` method. [[1]](diffhunk://#diff-6baf451de6db548637aa10622a7799cfdc2968b760f9c8a0f62a933e96870206L5-R33) [[2]](diffhunk://#diff-c5cb059dd021b182a095e078ed45b7ad55e6be02fc670dd560c7cc2c344ae14bL5-R33) [[3]](diffhunk://#diff-7b50f8df1a6b859e66cb9fa1c76630917f8c594236e5738e7cc8f633f7952105R16-R33)

Dependency and configuration updates:

* Upgraded `github.com/DATA-DOG/go-sqlmock` dependency from v1.5.0 to v1.5.2 in `go.mod` for improved testing support.
* Enabled `issue-845-fix` in `.mockery.yaml` as per the warning message.
* Removed the `go-consistent` tool for two major reasons:
  * part of its functionality is already covered by `golangci-lint`
  * it is inflexible and difficult to configure, for example it can't reliably skip tests, which is annoying

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -115,22 +115,6 @@ jobs:
           golangci_lint_flags: "-c=.golangci.yml"
           golangci_lint_version: v2.6.2 # Version should match specified in Makefile
 
-      - name: Run go-consistent
```
Comment (ademidoff) [inline .github/workflows/main.yml:118]: I'm removing the go-consistent tool for two major reasons:
* part of its functionality is already covered by golangci-lint
* it is inflexible and difficult to configure, for example it can't reliably skip tests, which requires workarounds and is simply annoying

File: managed/services/agents/handler.go [hunk 2]
```diff
@@ -227,8 +260,7 @@ func updateAgentStatus(
 	l := logger.Get(ctx)
 	l.Debugf("updateAgentStatus: %s %s %d", agentID, status, listenPort)
 
-	agent := &models.Agent{AgentID: agentID}
-	err := q.Reload(agent)
+	agent, err := models.FindAgentByID(q, agentID)
```
Comment (ademidoff) [inline managed/services/agents/handler.go:262]: Refactored in favor of using the model.

File: .mockery.yaml [hunk 3]
```diff
@@ -2,6 +2,7 @@
 disable-version-string: True
 with-expecter: False
 resolve-type-alias: False
+issue-845-fix: True
```
Comment (ademidoff) [inline .mockery.yaml:5]: Fixed a mockery warning.

**Review by JiriCtvrtka — APPROVED** [review]

File: managed/services/agents/deps.go [hunk 4]
```diff
@@ -32,6 +32,9 @@ import (
 // FIXME Rename to victoriaMetrics.Service, update tests.
 type prometheusService interface {
 	RequestConfigurationUpdate()
+	// ForceConfigurationUpdate triggers immediate synchronous configuration update,
+	// bypassing the batch delay. Use this for critical updates like port changes.
+	ForceConfigurationUpdate(ctx context.Context) error
```
Comment (maxkondr) [inline managed/services/agents/deps.go:40]: why a separate method is required? Can't port change be handled via RequestConfigurationUpdate?

Comment (ademidoff) [reply managed/services/agents/deps.go:40]: This one is synchronous, and `RequestConfigurationUpdate` is async. We previously used that one, and it worked well, except in cases where the port number is changed. In such case, we need to synchronously update VM's configuration, otherwise there will be bugs like the one in question.
